src/Symfony/Component/Translation/Writer/TranslationWriter.php

```diff
@@ -45,7 +46,9 @@ public function addDumper($format, DumperInterface $dumper)
     public function disableBackup()
     {
         foreach ($this->dumpers as $dumper) {
-            $dumper->setBackup(false);
+            if ($dumper instanceof FileDumper) {
```

Comment:
What about using `method_exists()` here instead to support users who create their own dumpers?